src/config.rs

```diff
@@ -2,7 +2,8 @@ use crate::ffi;
 
 /// Error correction mode
 #[derive(Clone, Copy)]
-#[repr(u32)]
+#[cfg_attr(target_os = "windows", repr(i32))]
+#[cfg_attr(not(target_os = "windows"), repr(u32))]
```

Comment:
Actually I would not like do the representation to be platform dependent.
I think more proper solution is type casting for ffi calls and back.
For example, `into_raw` function may cast `u32` to `c_int`.